### PR TITLE
[ERTE-5] feat: Added enterprise uuid support in course enrollment

### PR DESCRIPTION
**Pull request by rehanedly**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @rehanedly! I've created [OSPR-5896](https://openedx.atlassian.net/browse/OSPR-5896) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@rehanedly thank you for your contribution. @ziafazal Will you review this?

**Comment by ziafazal** [comment]
@natabene Can someone from enterprise team have a look at this?

File: enterprise/models.py
Comment (iloveagent57) [inline enterprise/models.py:1025]: [style nitpick] Could you move the first line of args onto their own line so the indentation adheres to our style guidelines?   Something like this:
```python
enrollment_api_client.enroll_user_in_course(
    self.username, 
    course_run_id, 
    mode, 
    cohort=cohort,
    enterprise_uuid=str(self.enterprise_customer.uuid),
)
```
https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines

File: enterprise/utils.py
Comment (iloveagent57) [inline enterprise/utils.py:1424]: [style nitpick] Same long-line breaking request as above.

[style nitpick] The `cohort` parameter defaults to `None`, so you don't have to include it in the call, here.

[Fix] I think you want
```python
enterprise_uuid=str(enterprise_customer_user.enterprise_customer.uuid)
```
here - the `EnterpriseCustomerUser` model doesn't have a `uuid` field, so you can drop the conditional.

File: enterprise/views.py
Comment (iloveagent57) [inline enterprise/views.py:587]: [style nitpick] If the args make this line too long, write it like
```python
existing_enrollment = enrollment_api_client.get_course_enrollment(
    request.user.username, course_id
)
```

File: enterprise/views.py
Comment (iloveagent57) [inline enterprise/views.py:1255]: [style nitpick] No need to include `cohort=None` here.

File: tests/test_enterprise/api_client/test_lms.py
Comment (iloveagent57) [inline tests/test_enterprise/api_client/test_lms.py:92]: [style nitpick] Move args to a new line as mentioned above.

File: enterprise/api_client/lms.py
Comment (iloveagent57) [inline enterprise/api_client/lms.py:237]: [mild suggestion] You could cast the `enterprise_uuid` to a `str()` on this line.  Then the caller doesn't have to worry about passing in a UUID object vs. a string, since the `post()` call will "do the right thing".

**Review by iloveagent57 — CHANGES_REQUESTED** [review]
Thanks, looks pretty good overall!

File: enterprise/views.py
Comment (alex-sheehan-edx) [inline enterprise/views.py:1648]: [style nitpick] is there a reason for the spacing being added here?  Could we keep the comment above the code?

File: tests/test_enterprise/test_signals.py
Comment (alex-sheehan-edx) [inline tests/test_enterprise/test_signals.py:174]: [style nitpick] I think this can be moved to multiple lines as Alex D has mentioned elsewhere in this PR.

**Review by alex-sheehan-edx — APPROVED** [review]
I agree with all of @iloveagent57's comments but otherwise looks pretty good to me.

File: tests/test_enterprise/test_signals.py
Comment (rehanedly) [reply tests/test_enterprise/test_signals.py]: @alex-sheehan-edx It's done

File: enterprise/views.py
Comment (rehanedly) [reply enterprise/views.py]: @alex-sheehan-edx It's done

File: enterprise/models.py
Comment (rehanedly) [reply enterprise/models.py]: @iloveagent57  It's done

File: enterprise/utils.py
Comment (rehanedly) [reply enterprise/utils.py]: @iloveagent57  It's done

File: enterprise/views.py
Comment (rehanedly) [reply enterprise/views.py]: @iloveagent57  It's done

File: enterprise/views.py
Comment (rehanedly) [reply enterprise/views.py]: @iloveagent57  It's done

File: tests/test_enterprise/api_client/test_lms.py
Comment (rehanedly) [reply tests/test_enterprise/api_client/test_lms.py]: @iloveagent57  It's done

File: enterprise/api_client/lms.py
Comment (rehanedly) [reply enterprise/api_client/lms.py]: @iloveagent57  It's done

**Review by iloveagent57 — APPROVED** [review]

**Comment by iloveagent57** [comment]
@rehanedly There are some more test assertions to fix in `tests/test_admin/test_view.py`

**Comment by natabene** [comment]
@iloveagent57 Can you merge this once this is good? @ziafazal doesn't have merge rights in the repo, I believe.

**Comment by edx-webhook** [comment]
@rehanedly 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-enterprise&entry.867055334=https://github.com/edx/edx-enterprise/pull/1288&entry.1484655318=https://github.com/rehanedly&entry.752974735=2021-07-01+10:21&entry.1917517419=2021-07-14+17:28&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
